haskell-src/exec/Chainweb/Gaps.hs

```diff
@@ -80,12 +87,108 @@ gapsCut env args cutBS = do
     f :: LogFunctionIO Text -> IORef Int -> Int64 -> (Low, High) -> IO ()
     f logger count cid (l, h) = do
       let range = (ChainId (fromIntegral cid), l, h)
-      headersBetween env range >>= \case
-        Left e -> logger Error $ fromString $ printf "ApiError for range %s: %s" (show range) (show e)
+      let onCatch (e :: SomeException) = case errorLogFile of
+            Nothing -> do
+              logger Error $ fromString $ printf "Caught exception from headersBetween for range %s: %s" (show range) (show e)
+              pure $ Right []
+            Just fp -> withFileLogger fp Error $ do
+              S.logg Error $ fromString $ printf "Caught exception from headersBetween for range %s: %s" (show range) (show e)
+              pure $ Right []
```

Comment:
I see a lot of code duplication of this kind, would be nice to deduplicate them (if we're planning to keep the errorLogFile)